scripts/pages/pgCategories.ts

```diff
@@ -43,16 +45,29 @@ export default class PgCategories extends withDismissAndBackButton(PgCategoriesD
         this.categoriesGrid.refreshData();
     }
     async fetchCategories() {
-        this.categories = await getCategories();
-        if (this.categories) {
-            this.refreshGridView();
+        this.waitDialog.show();
+        try {
+            this.categories = await getCategories();
+            if (this.categories) {
+                this.refreshGridView();
+            }
+        } catch (error) {
+        } finally {
+            setTimeout(() => {
+                this.waitDialog.hide();
+            }, 1000);
         }
     }
+    initDialog() {
+        const flWaitDialog = new FlWaitDialog();
+        this.waitDialog = dialog(flWaitDialog);
```

Comment:
you can set this on constructor like this 1 liner

this.waitDialog = dialog(new FlWaitDialog());